boards: atsame54_xpro: Add SAM E54 Xplained Pro Evaluation Kit

benpicco opened the pull request:
This adds support for the Atmel SAM E54 Xplained Pro Evaluation Kit.

split off from #14685
depends on ~~#20062~~  ~~#20056~~ ~~#20058~~
additional functionality depends on ~~#20052~~ ~~#20055~~ ~~#20057~~ ~~#21866~~

benpicco committed:
boards: atsame54_xpro: Add SAM E54 Xplained Pro Evaluation Kit

This adds support for the Atmel SAM E54 Xplained Pro Evaluation Kit.

Only basic functionality has been tested so far.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

dbkinder reviewed (approved):
doc change LGTM, thanks.

galak reviewed (changes requested):
Can you shrink the images, they are huge (file size).

Does this only need the SoC PR, or do we need UART driver?  GPIO driver?  [ what's the minimal set of SoC + drivers to make the board code useful ].

benpicco commented:
Down to
```
808K ATSAME54-XPRO-pinout.png 
596K atsame54_xpro.png
```

If you just want to blink an LED, the SoC PR is enough.
The GPIO PR is about EXTI (and port D), basic GPIO functionality is the same as on samd2x.

UART would be nice of course. :wink:

galak commented:
> Down to
> 
> ```
> 808K ATSAME54-XPRO-pinout.png 
> 596K atsame54_xpro.png
> ```
> 
> If you just want to blink an LED, the SoC PR is enough.
> The GPIO PR is about EXTI (and port D), basic GPIO functionality is the same as on samd2x.
> 
> UART would be nice of course. 😉

Sent you on slack a much smaller image, take a look and see what you think.

galak commented:
@benpicco can you look at addressing the build issues / sanitycheck failures.  If you want this in for 2.1 we need to resolve this by friday (Nov 8th).

benpicco committed:
dts: atmel_sam0: fix location of flash and sram blocks

The blocks were moved into the soc block in samd5x.dtsi,
so we also have to move them for the the actual SoC definitions
that inherit from that.

Signed-off-by: Benjamin Valentin <benjamin.valentin@ml-pa.com>

benpicco committed:
soc: atmel_sam0: remove obsolete include

`cortex_m/exc.h` is not needed anymore and ceased to exist.

Signed-off-by: Benjamin Valentin <benjamin.valentin@ml-pa.com>

benpicco commented:
I forgot to also move the `flash` and `sram` blocks in the `samd5xx20.dtsi` etc DTS files after moving it in `samd5x.dts` - this is fixed by #20420 

Then the DTS processor will complain about the unknown lock regions property introduced by #20058 - you'll get past the config with this PR 😉 

Then `soc.h` wants to include `tc_fixup_samd5x.h` which is introduced by #20056

With these three, `hello_world` compiles.

ioannisg commented:
> This adds support for the Atmel SAM E54 Xplained Pro Evaluation Kit.
> 
> split off from #14685
> depends on #20062

#20062 is merged; update the description :)

ioannisg commented on boards/arm/atsame54_xpro/Kconfig.board:
```diff
@@ -0,0 +1,8 @@
+# Kconfig - SAM E54 Xplained Pro Board configuration
```
Remove Kconfig, start with "SAM E54 ..."

ioannisg replied:
And the same for all these license headers :)

ioannisg commented on tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf:
```diff
@@ -0,0 +1,2 @@
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_0"
```
Why is this change here?

benpicco replied:
So you can run the `spi_loopback` test on this board - other boards [have it too](https://github.com/zephyrproject-rtos/zephyr/tree/master/tests/drivers/spi/spi_loopback/boards/)

ioannisg replied:
Sure, I was wondering "why part of this PR" - is the SPI driver already merged?

benpicco replied:
Alright I moved it to the SPI PR.

ioannisg reviewed (changes requested):
The test file looks not relevant to the PR.
Some clean up of the Kconfig file headers is needed.
Looking good, otherwise

galak commented:
@benpicco still major CI failures

benpicco commented:
@galak I removed references the not yet merged features.

benpicco commented:
If the other PRs get merged before this, I will remove the last few commits here.

ioannisg commented:
@benpicco can you address the checkpatch warnings?

benpicco commented:
@ioannisg I'm not sure how - I can't find documentation for the vendor strings in [`atsamd21_xpro.dts`](https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/atsamd21_xpro/atsamd21_xpro.dts#L12) either - where should they go?

benpicco committed:
soc: atmel_sam0: remove CONFIG_ENTROPY_NAME from dts_fixup.h

It's not needed anymore and causes build failures.

Signed-off-by: Benjamin Valentin <benjamin.valentin@ml-pa.com>

ioannisg commented:
@galak can you take a look?

benpicco commented:
Thank you for the review & merge!
